nssp_demo/forecast_state.py

```diff
@@ -0,0 +1,222 @@
+import argparse
+import logging
+import os
+import subprocess
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import numpyro
+import polars as pl
+from prep_data import process_and_save_state
+
+numpyro.set_host_device_count(4)
+
+from fit_model import fit_and_save_model  # noqa
+from generate_predictive import generate_and_save_predictions  # noqa
+
+
+def forecast_denominator(model_dir: Path, n_forecast_days: int) -> None:
```

Comment:
Since this is run in the same place as the numerator forecasting, we should revise to guarantee that they produce the same number of samples.

Reply:
Since we're mixing languages, I wonder if you could just do an inline Rscript to detect the devices with [`parallelly::availableCores`](https://parallelly.futureverse.org/reference/availableCores.html).

Reply:
But this is a demo, so I would just pick 2 and leave that as a default with `npus = 2 # set this accordingly` or whatnot.

Reply:
To discuss in #85

Reply:
Somehow this thread got way of track from my original comment, which, I believe, remains unaddressed.

Reply:
Original request addressed in 8062222